src/commands/redis_cmd.cc

```diff
@@ -1508,25 +1508,60 @@ class CommandHGetAll : public Commander {
 class CommandHRange : public Commander {
  public:
   Status Parse(const std::vector<std::string> &args) override {
-    if (args.size() != 6 && args.size() != 4) {
-      return Status(Status::RedisParseErr, errWrongNumOfArguments);
-    }
-    if (args.size() == 6 && Util::ToLower(args[4]) != "limit") {
-      return Status(Status::RedisInvalidCmd, errInvalidSyntax);
+    CommandParser parser(args, 4);
+    while (parser.Good()) {
+      if (parser.EatEqICaseFlag("BYLEX", by_flag_)) {
+      } else if (parser.EatEqICase("REV")) {
+        reversed_ = true;
+      } else if (parser.EatEqICase("LIMIT")) {
+        offset_ = GET_OR_RET(parser.TakeInt());
+        count_ = GET_OR_RET(parser.TakeInt());
+      } else {
+        return parser.InvalidSyntax();
+      }
     }
-    if (args.size() == 6) {
-      auto parse_result = ParseInt<int64_t>(args_[5], 10);
-      if (!parse_result) return Status(Status::RedisParseErr, errValueNotInteger);
-      limit_ = *parse_result;
+    if (by_flag_ == "BYLEX") {
+      spec_.reversed = reversed_;
+      spec_.count = count_;
+      spec_.offset = offset_;
+      Status s;
+      if (spec_.reversed) {
+        s = Redis::Hash::ParseRangeLexSpec(args[3], args[2], &spec_);
+      } else {
+        s = Redis::Hash::ParseRangeLexSpec(args[2], args[3], &spec_);
+      }
+      if (!s.IsOK()) {
+        return Status(Status::RedisParseErr, s.Msg());
+      }
+    } else {
+      by_flag_ = "BYINDEX";
```

Comment:
BYINDEX should make no sense for the hash?

Reply:
Maybe there is a need for a minimum of k values or something similar?

Reply:
For example, to take the smallest element, you can use `hrange 0 0 ` or `hrange - + limit 0 1`. You do not need to specify a limit when using index, just the desired range.

Reply:
@tanruixiang What do you mean by "the smallest element in a hash"?

Reply:
> @tanruixiang What do you mean by "the smallest element in a hash"?

The dictionary order in hash is minimum.

Reply:
I can get your point, but "the smallest element in a hash" is implicitly in LSM storage and may be confusing. IMHO, supporting the BYLEX should be enough for the hash range.

Reply:
Yes, they can do the same thing, I'll change the command to `hrangebylex`.

Reply:
Yes, I think it's good to keep the hrange command with BYLEX only.